### PR TITLE
[FIX] Fisherman getting paid for fish even if mcMMO fishing protection kicks in

### PR DESCRIPTION
**Pull request by icarapovic**
When the mcMMO fishing protections are on, and they are triggered, even though mcMMO prevents the player from obtaining fish and XP, Jobs will still award money, points and XP even if no fish/xp is given to the player. I tried to fix this by copying over the mcMMO checks logic into Jobs.

The issue I encountered was that somehow **the version of mcMMO .jar in Jobs/libs was 2.1.2** while **the one I downloaded directly from mcMMO repository was 2.1.175** (-SNAPSHOT) when I build it. This replacement was required because the classes that were performing the checks were not available inside the 2.1.2 jar, therefore the entire thing would not work / be possible. 
Tbh I have no idea how did Jobs acquire a version that is supposedly newer then the latest version of the actual mcMMO.

Since this is my first attempt at MC Mod development, I'm pretty sure this probably not a proper solution but at best a "good temporary fix". If you can use this PR as a base to implement this correctly, feel free to do so and discard my PR when done. 

Anyways, feedback is appreciated.

### CODE REVIEW

**Review by montlikadani — CHANGES_REQUESTED** [review]

File: src/main/java/com/gamingmesh/jobs/listeners/JobsPaymentListener.java [hunk 1]
```diff
@@ -490,6 +493,12 @@ public void onPlayerFish(PlayerFishEvent event) {
 	    return;
 
 	if (event.getState() == PlayerFishEvent.State.CAUGHT_FISH && event.getCaught() instanceof Item) {
+		if(ExperienceConfig.getInstance().isFishingExploitingPrevented()) {
+			if (UserManager.getPlayer(player).getFishingManager().isExploitingFishing(event.getHook().getLocation().toVector())) {
+				return;
+			}
+		}
```
[the remark this answers is not in the document]
Comment (icarapovic) [reply src/main/java/com/gamingmesh/jobs/listeners/JobsPaymentListener.java]: Updated.

File: src/main/java/com/gamingmesh/jobs/listeners/JobsPaymentListener.java [hunk 2]
```diff
@@ -490,6 +493,12 @@ public void onPlayerFish(PlayerFishEvent event) {
 	    return;
 
 	if (event.getState() == PlayerFishEvent.State.CAUGHT_FISH && event.getCaught() instanceof Item) {
+		if(ExperienceConfig.getInstance().isFishingExploitingPrevented()) {
+			if (UserManager.getPlayer(player).getFishingManager().isExploitingFishing(event.getHook().getLocation().toVector())) {
```
[the remark this answers is not in the document]
Comment (icarapovic) [reply src/main/java/com/gamingmesh/jobs/listeners/JobsPaymentListener.java]: Updated.

File: src/main/java/com/gamingmesh/jobs/listeners/JobsPaymentListener.java [hunk 3]
```diff
@@ -490,6 +493,12 @@ public void onPlayerFish(PlayerFishEvent event) {
 	    return;
 
 	if (event.getState() == PlayerFishEvent.State.CAUGHT_FISH && event.getCaught() instanceof Item) {
+		if(ExperienceConfig.getInstance().isFishingExploitingPrevented()) {
```
[the remark this answers is not in the document]
Comment (icarapovic) [reply src/main/java/com/gamingmesh/jobs/listeners/JobsPaymentListener.java]: Updated.

**Comment by icarapovic** [comment]
I've updated the formatting as requested. Did you check the mcMMO .jar? The used jar is a snapshot version I built myself from the official sources, so it may not be the one you would want to use. However, since that .jar that is currently used by Jobs does not contain the checks, it definitely needs to be updated in order for this to work.

**Comment by montlikadani** [comment]
Yep, I checked the jar file if you're forked the new [mcMMO](https://github.com/mcMMO-Dev/mcMMO) and compiled a new jar file, then we're good, as we uses this project.

**Review by montlikadani — APPROVED** [review]
LGTM

**Pull request merged**